tests/gradients/parameter_shift/test_parameter_shift.py

```diff
@@ -1567,6 +1567,27 @@ def test_single_expectation_value(self, tol):
         assert np.allclose(res[0], expected[0], atol=tol, rtol=0)
         assert np.allclose(res[1], expected[1], atol=tol, rtol=0)
 
+    @pytest.mark.parametrize(
+        "par", [0, 1, 2, 3, np.int8(1), np.int16(1), np.int32(1), np.int64(1)]
+    )  # integers, zero
+    def test_integer_parameters(self, tol, par):
+        """Test that the gradient of the RY gate matches the exact analytic formula."""
+        dev = qml.device("default.qubit", wires=2)
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RY(par, wires=[0])
+            qml.expval(qml.PauliX(0))
+
+        tape.trainable_params = {0}
```

Comment:
```suggestion
        tape = qml.tape.QuantumScript([qml.RY(par, wires=[0])], [qml.expval(qml.PauliX(0))])
        tape.trainable_params = {0}
```

Unless we are actually testing something involving queuing, I prefer to use the "construct on initialization" syntax. No need for a mutable state and a global interpreter lock this way.